pkg/collector/corechecks/snmp/internal/checkconfig/config.go

```diff
@@ -733,8 +733,13 @@ func GetProfileForSysObjectID(profiles profileConfigMap, sysObjectID string) (st
 			if !found {
 				continue
 			}
-			if matchedProfile, ok := tmpSysOidToProfile[oidPattern]; ok {
-				return "", fmt.Errorf("profile %s has the same sysObjectID (%s) as %s", profile, oidPattern, matchedProfile)
+			if prevMatchedProfile, ok := tmpSysOidToProfile[oidPattern]; ok {
+				if profiles[prevMatchedProfile].isUserProfile && !profConfig.isUserProfile {
```

Comment:
Trying to make sure I read this right, but it seems to me like this will give precedence to _any_ user profile versus our profiles.
Meaning that if they have a simple "base" profile, matching on `1.3.6.1.4.*` all devices will start to use this profile instead.
I thought we wanted to give precedence to custom profiles only when they match on the same sysObjectID

Reply:
The way it's currently implemented, the 1/ merge of `profiles` and `default_profiles` and the 2/ sysObjectID matching are two different steps.

In step 1/ when merging process doesn't rely on the sysObjectID. And the sysObjectID is currently not inherited, only the sysObjectID of the leaf profile is considered. If the user profile override a base profile that `1.3.6.1.4.*`, the user profile will take precedence the sysObjectID of the user profile will be used.
In case the user want to use a different sysObjectID and not override, maybe it's easier to just create a profile with different name and extends the "base" profile with `1.3.6.1.4.*`.

> I thought we wanted to give precedence to custom profiles only when they match on the same sysObjectID

We give precedent to user profile when the same sysObjectID is present multiple profiles in `profiles` and `default_profiles`.

Reply:
I'm really talking about step 2/ here. My example is let's say a user has a single custom profile `/snmp.d/profiles/my-custom-profile.yaml` with the following:

```yaml
sys_object_ids:
  - 1.*

metrics:
 ...
```

What I understand is that this custom profile will take precedence over all datadog OOTB profiles and I'm wondering if that's the intention

Reply:
> What I understand is that this custom profile will take precedence over all datadog OOTB profiles and I'm wondering if that's the wanted behavior

Hum, that's actually not the case. Added this additional unit test to cover this exact case: https://github.com/DataDog/datadog-agent/pull/17395/commits/1961c671f45f004ee2ff85e688a923a2d68b45ac

(maybe I'm misunderstanding your point)

@FlorianVeaux

Reply:
Note: the variable `tmpSysOidToProfile` is a simple `set` of sysObjectIDs like `{"1.*", "1.2.3"}` and in your example, if `profiles/my-custom-profile.yaml` has sysobjectid: 1.*, and `default_profiles/some_profile.yaml` has "1.2.3".

At this point of the code, `tmpSysOidToProfile` will have both entries.

The resolution is done few lines later with `getMostSpecificOid(matchedOids)`.

Reply:
Makes sense, thanks for explaining and adding the unit test!